### PR TITLE
bluetooth: remove support for 3-wire bt coex

### PR DESCRIPTION
**Pull request by martintv**
3-wire bt coex is going to be replaced by the new
generic mpsl_cx. So remove the sample and support for the the old one.

### CODE REVIEW

File: doc/nrf/ug_bt_coex.rst [hunk 1]
```diff
@@ -7,63 +7,7 @@ Using Bluetooth external radio coexistence
...
-To do so, use the HCI VS command :c:enum:`SDC_HCI_OPCODE_CMD_VS_COEX_SCAN_MODE_CONFIG`.
-Its parameters are described in  :c:type:`sdc_hci_cmd_vs_coex_scan_mode_config_t`.
-
-.. _ug_bt_coex_3w_sample:
-
-Sample application
-******************
-
-A sample application for 3-wire coexistence can be found at :file:`samples/bluetooth/radio_coex_3wire`.
+The 3-Wire coexistence protocol has been removed, use the genric mspl_cx instead
```
Comment (jciupis) [inline doc/nrf/ug_bt_coex.rst:10]: Should there be a mention in the changelog/release notes that support for 3-wire BT coexistence is removed?

File: subsys/mpsl/cx/CMakeLists.txt [hunk 2]
```diff
@@ -4,11 +4,10 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-if (CONFIG_MPSL_CX_THREAD OR CONFIG_MPSL_CX_BT_1WIRE OR CONFIG_MPSL_CX_BT_3WIRE OR CONFIG_MPSL_CX_NRF700X)
+if (CONFIG_MPSL_CX_THREAD OR CONFIG_MPSL_CX_BT_1WIRE  OR CONFIG_MPSL_CX_NRF700X)
     zephyr_library()
 
     zephyr_library_sources_ifdef(CONFIG_MPSL_CX_THREAD thread/mpsl_cx_thread.c)
     zephyr_library_sources_ifdef(CONFIG_MPSL_CX_BT_1WIRE bluetooth/mpsl_cx_1w_bluetooth.c)
-    zephyr_library_sources_ifdef(CONFIG_MPSL_CX_BT_3WIRE bluetooth/mpsl_cx_3w_bluetooth.c)
```
Comment (jciupis) [inline subsys/mpsl/cx/CMakeLists.txt:12]: Is it intentional that you have not removed the source file `mpsl_cx_3w_bluetooth.c`?

File: subsys/mpsl/cx/CMakeLists.txt [hunk 3]
```diff
@@ -4,11 +4,10 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-if (CONFIG_MPSL_CX_THREAD OR CONFIG_MPSL_CX_BT_1WIRE OR CONFIG_MPSL_CX_BT_3WIRE OR CONFIG_MPSL_CX_NRF700X)
+if (CONFIG_MPSL_CX_THREAD OR CONFIG_MPSL_CX_BT_1WIRE  OR CONFIG_MPSL_CX_NRF700X)
```
Comment (jciupis) [inline subsys/mpsl/cx/CMakeLists.txt:7]: Does it make sense to also remove the Kconfig option `MPSL_CX_BT_3WIRE` from `mpsl/cx/Kconfig`?

**Review by KAGA164 — APPROVED** [review]
Please update changelog

**Pull request closed without merging**